packages/mui-material/src/ButtonGroup/ButtonGroup.js

```diff
@@ -109,19 +109,25 @@ const ButtonGroupRoot = styled('div', {
       }),
       ...(ownerState.variant === 'text' &&
         ownerState.orientation === 'horizontal' && {
-          borderRight: `1px solid ${
-            theme.palette.mode === 'light' ? 'rgba(0, 0, 0, 0.23)' : 'rgba(255, 255, 255, 0.23)'
-          }`,
+          borderRight: theme.vars
+            ? `1px solid rgba(${theme.vars.palette.common.onBackgroundChannel} / 0.23)`
+            : `1px solid ${
+                theme.palette.mode === 'light' ? 'rgba(0, 0, 0, 0.23)' : 'rgba(255, 255, 255, 0.23)'
+              }`,
         }),
       ...(ownerState.variant === 'text' &&
         ownerState.orientation === 'vertical' && {
-          borderBottom: `1px solid ${
-            theme.palette.mode === 'light' ? 'rgba(0, 0, 0, 0.23)' : 'rgba(255, 255, 255, 0.23)'
-          }`,
+          borderBottom: theme.vars
+            ? `1px solid rgba(${theme.vars.palette.common.onBackgroundChannel} / 0.23)`
+            : `1px solid ${
+                theme.palette.mode === 'light' ? 'rgba(0, 0, 0, 0.23)' : 'rgba(255, 255, 255, 0.23)'
+              }`,
```

Comment:
@vicasas I have added
- `common.backgroundChannel`
- `common.onBackgroundChannel`

in https://github.com/mui/material-ui/pull/32128.

I think it makes sense to use it here.

Reply:
@siriwatknp Perfect! I'll check back later to make the change.

Reply:
I think I misunderstood the comment at first. 😅

`backgroundChannel` and `onBackgroundChannel` returns a color for dark/light?

Reply:
@vicasas yes.

light mode, `--md-palette-common-backgroundChannel: 255 255 255;`.
dark mode, `--md-palette-common-backgroundChannel: 0 0 0;`.

Reply:
Great!  now it makes sense to me